app/Main.hs

```diff
@@ -290,8 +288,8 @@ main = do
         , userGPG = oGPGSign
         }
   case (ms3Bucket, oAwsAccessKey, oAwsSecretKey) of
-    (Just _, Just accessKey, Nothing) -> setEnv "AWS_ACCESS_KEY_ID" accessKey
-    (Just _, Nothing, Just secretKey) -> setEnv "AWS_SECRET_KEY_ID" secretKey
+    (Just _, Just _, Nothing) -> error $ "--aws-secret-key is also required."
```

Comment:
Just to make sure we are on the same page, here are all possible options regarding AWS uploading:

* s3 bucket name is provided though CLI or the variable `S3_BUCKET` is set:
  * `aws-secret-key` or `aws-access-key` is provided but not both: errors out with a missing argument message
  * neither of `aws-secret-key` and `aws-access-key` is provided: uses environment variables, and if those are missing `amazonka` will error out.
* s3 bucket name is NOT provided though CLI and the variable `S3_BUCKET` is NOT set:
  * `aws-secret-key` or `aws-access-key` is provided but not both: errors out with missing argument message (I think it's an expected behavior, since none of the keys should be supplied without a bucket name anyways)
  * a warning is displayed that uploading is turned off, regardless of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_KEY_ID` variables.